sklearn/preprocessing/_target_encoder.py

```diff
@@ -234,24 +235,45 @@ def fit_transform(self, X, y):
                 self.cv, shuffle=self.shuffle, random_state=self.random_state
             )
 
+        y_1d = y
+        n_classes = None
         X_out = np.empty_like(X_ordinal, dtype=np.float64)
         X_unknown_mask = ~X_known_mask
-
-        for train_idx, test_idx in cv.split(X, y):
+        X_ordinal_trans = X_ordinal
+        if self.target_type_ == "multiclass":
+            y_1d = self._label_binarizer_.inverse_transform(y)
+            n_classes = self._label_binarizer_.classes_.shape[0]
+            X_out, X_ordinal_trans, X_unknown_mask = [
```

Comment:
I do not think we need to do `np.repeat` on `X_oridinal_trans` or `X_unknown_mask` and define `X_out` as follows:

```python
X_out = np.empty(
    (X_ordinal.shape[0], X_ordinal.shape[1] * n_classes), dtype=np.float64
)
```

then have a custom `_transform_X_multiclass` that knows how to handle them.

Reply:
This is so much better and cleaner. I have not made another `_transform_X_multiclass` as I thought it would be messy adding another 'if multiclass' to the transform methods and instead used `_transform_X_ordinal` for both. Happy to change though.